### PR TITLE
feat(cb2-8666, cb2-8806, cb2-8665, cb2-8089, cb2-8667): update vtm frontend to use v3 api

### PR DESCRIPTION
**Pull request by gjulien-bjss**
## Update vtm frontend to use v3 api

Update vtm frontend to use v3 api for psv, trl, hgv, light vehicles.

[CB2-8666](https://dvsa.atlassian.net/browse/CB2-8666)
[CB2-8806](https://dvsa.atlassian.net/browse/CB2-8806)
[CB2-8665](https://dvsa.atlassian.net/browse/CB2-8665)
[CB2-8667](https://dvsa.atlassian.net/browse/CB2-8667)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/features/tech-record/create-batch/components/batch-vehicle-template/batch-vehicle-template.component.spec.ts [hunk 1]
```diff
@@ -83,7 +84,7 @@ describe('BatchVehicleTemplateComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
-
+  // TODO V3 HGV PSV TRL
```
Comment (naathanbrown) [inline src/app/features/tech-record/create-batch/components/batch-vehicle-template/batch-vehicle-template.component.spec.ts:87]: This needs removed

File: src/app/features/tech-record/create/components/hydrate-new-vehicle-record/hydrate-new-vehicle-record.component.spec.ts [hunk 2]
```diff
@@ -66,7 +68,7 @@ describe('HydrateNewVehicleRecordComponent', () => {
 
       expect(clearErrorsSpy).toHaveBeenCalledTimes(1);
     });
-
+    // TODO V3 HGV PSV TRL
```
Comment (naathanbrown) [inline src/app/features/tech-record/create/components/hydrate-new-vehicle-record/hydrate-new-vehicle-record.component.spec.ts:71]: same

File: src/app/forms/custom-sections/plates/plates.component.spec.ts [hunk 3]
```diff
@@ -62,46 +63,44 @@ describe('PlatesComponent', () => {
...
+          plateReasonForIssue: 'Replacement'
+        }
+      ];
       const plateFetched = component.mostRecentPlate;
 
       expect(plateFetched).toBeDefined();
       expect(plateFetched!.plateSerialNumber).toEqual('123456');
     });
 
+    //TODO: Remove the anys
```
Comment (naathanbrown) [inline src/app/forms/custom-sections/plates/plates.component.spec.ts:80]: remove TODO

File: src/app/forms/custom-sections/psv-brakes/psv-brakes.component.ts [hunk 4]
```diff
@@ -42,22 +43,28 @@ export class PsvBrakesComponent implements OnInit, OnChanges, OnDestroy {
   ngOnInit(): void {
     this.form = this.dfs.createForm(PsvBrakesTemplate, this.vehicleTechRecord) as CustomFormGroup;
 
+    //TODO: remove the anys
```
Comment (naathanbrown) [inline src/app/forms/custom-sections/psv-brakes/psv-brakes.component.ts:46]: again

**Review by naathanbrown — APPROVED** [review]

**Review by naathanbrown — APPROVED** [review]

**Review by owen-corrigan-bjss — APPROVED** [review]

**Review by owen-corrigan-bjss — DISMISSED** [review]

**Review by owen-corrigan-bjss — DISMISSED** [review]

**Review by shivangidas — APPROVED** [review]

**Review by owen-corrigan-bjss — APPROVED** [review]

**Review by shivangidas — APPROVED** [review]

**Pull request merged**